api-documentation.md

```diff
@@ -203,6 +206,16 @@ When the necessary demographics are not know for a specific device id (a user) t
 
 If the **"content_type"** parameter is set to **"html"** or is ommited then the request will be served like an ordinary offerwall request, returning an html page with the surveys matched for each device_id (user).
 
+
+### Notes for `sig` query parameter
+In order to prevent any tampering on the `reward_conversion` and `reward_name` parameters, the platform supports validation by requiring a hash of the `reward_conversion`, `reward_name`, and `click_id` values passed on `json` parameter on register device. You can sign the combination of `reward_conversion`+`reward_name`+`click_id` parameter using the [HMAC-SHA1](https://en.wikipedia.org/wiki/HMAC) algorithm and your account's secret_key that can be retrieved from the [Account Information](//www.pollfish.com/dashboard/dev/account/information) page. The value should be then encoded using [Base64](https://en.wikipedia.org/wiki/Base64), then URL encoded using [Percent encoding](https://en.wikipedia.org/wiki/Percent-encoding) and provided in the `reward_conversion_hash` parameter. Note that when providing `reward_conversion` and we have enabled the hashing option on your app `sig` must be set too.
```

Comment:
This param can be used optionally to prevent tampering around reward conversion hash param if passed within the register call. The platform supports url validation by requiring a hash of the `reward_conversion`, `reward_name`, and `click_id` values passed through the `json` parameter of the register device request. Failure to pass validation will return a status ...

In order to generate the sig field you should sign the combination of `reward_conversion`+`reward_name`+`click_id` parameter using the [HMAC-SHA1](https://en.wikipedia.org/wiki/HMAC) algorithm and your account's secret_key that can be retrieved from the [Account Information](//www.pollfish.com/dashboard/dev/account/information) page. 

The sig value should be then encoded using [Base64](https://en.wikipedia.org/wiki/Base64), then URL encoded using [Percent encoding](https://en.wikipedia.org/wiki/Percent-encoding) and provided in the `sig` parameter.